server/service/mdm.go

```diff
@@ -1116,6 +1117,12 @@ func (svc *Service) DeleteMDMWindowsConfigProfile(ctx context.Context, profileUU
 		return ctxerr.Wrap(ctx, err)
 	}
 
+	// prevent deleting Windows OS Updates profile (controlled by the OS Updates settings)
+	if _, ok := microsoft_mdm.FleetReservedProfileNames()[prof.Name]; ok {
+		err := &fleet.BadRequestError{Message: fmt.Sprintf("Profile name %q is reserved, it cannot be deleted.", prof.Name)}
```

Comment:
@marko-lisica or @noahtalerman similar case here, except this is when an attempt is made to delete a profile with the Fleet-reserved name.

Reply:
@mna Do we use a similar message for macOS profiles?

Reply:
I thought it would've been similar, but turns out it's "profiles managed by Fleet can't be deleted using this endpoint." for macOS, I'll update this message to be the same for Windows.

Reply:
Great, let's keep this one consistent with macOS as well.